app/presenters/order_detail_notice_presenter.rb

```diff
@@ -0,0 +1,85 @@
+class OrderDetailNoticePresenter < DelegateClass(OrderDetail)
+
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::OutputSafetyHelper
+
+  def statuses
+    [].tap do |statuses|
+      statuses << Notice.new(:in_review) if in_review?
+      statuses << Notice.new(:in_dispute) if in_dispute?
+      statuses << Notice.new(:can_reconcile) if can_reconcile_journaled?
+      statuses << Notice.new(:in_open_journal) if in_open_journal?
+    end
+  end
+
```

Comment:
I feel like that's almost too clever. I find that pretty hard to reason through, especially with the splats. Maybe not as elegant, I think the current way makes it pretty obvious what's happening without having to dive down the rabbit hole of the helper methods.

I personally like the `tap` version. The local variable version looks like a wall. Line breaks definitely make it better, but I kind of prefer the indentation. I'd be happy to switch if you definitely prefer it.

```
    statuses = []
    statuses << Notice.new(:in_review) if in_review?
    statuses << Notice.new(:in_dispute) if in_dispute?
    statuses << Notice.new(:can_reconcile) if can_reconcile_journaled?
    statuses << Notice.new(:in_open_journal) if in_open_journal?
    statuses
```